R/get_rcode.R

```diff
@@ -189,32 +118,83 @@ get_rcode <- function(datasets = NULL,
 
   # remove error with curly brace
   code_to_style <- gsub("}\n\\s*else", "} else", code_to_style)
+  code_to_style <- paste0(styler::style_text(code_to_style), collapse = "\n")
+  paste(code_not_to_style, code_to_style, sep = "\n")
+}
+
 
-  # return styler::style_text output - use progress in case shiny session is there
-  if (exists("session") && (!is.null(session$progressStack))) {
-    progress$set(message = "Getting R Code", value = 0.7)
+#' Get datasets code
+#'
+#' Get combined code from `FilteredData` and from `CodeClass` object.
+#' @param datanames (`character`) names of datasets to extract code from
+#' @param datasets (`FilteredData`) object
+#' @return `character(3)` containing following elements:
+#'  - code from `CodeClass` (data loading code)
+#'  - hash of loaded objects
+#'  - filter panel code
+#' @keywords internal
+get_datasets_code <- function(datanames, datasets) {
+  str_code <- datasets$get_code(datanames)
+  if (length(str_code) == 0 || (length(str_code) == 1 && str_code == "")) {
+    str_code <- paste0(c(
+      "#################################################################",
+      "# ___  ____ ____ ___  ____ ____ ____ ____ ____ ____ _ _  _ ____ #",
+      "# |__] |__/ |___ |__] |__/ |  | |    |___ [__  [__  | |\\ | | __ #",
+      "# |    |  \\ |___ |    |  \\ |__| |___ |___ ___] ___] | | \\| |__] #",
+      "#              _ ____    ____ _  _ ___  ___ _   _               #",
+      "#              | [__     |___ |\\/| |__]  |   \\_/                #",
+      "#              | ___]    |___ |  | |     |    |                 #",
+      "#################################################################\n"
+    ), collapse = "\n")
+  } else if (length(str_code) > 0) {
+    str_code <- paste0(str_code, "\n\n")
+  }
+  if (!datasets$get_check()) {
+    check_note_string <- paste0(
+      c(
+        "## NOTE: Reproducibility of data import and preprocessing was not",
+        "## explicitly checked (argument \"check = FALSE\" is set).",
+        "## The app developer has the choice to check the reproducibility",
+        "## and might have omitted this step for some reason. Please reach",
+        "## out to the app developer for details.\n"
+      ),
+      collapse = "\n"
+    )
+    str_code <- paste0(str_code, "\n\n", check_note_string)
   }
 
-  code_to_style <- paste0(styler::style_text(code_to_style), collapse = "\n")
+  str_hash <- paste(
+    paste0(
+      vapply(
+        datanames,
+        function(dataname) {
+          sprintf(
+            "# %s MD5 hash at the time of analysis: %s",
+            dataname,
+            datasets$get_filtered_dataset(dataname)$get_hash()
+          )
+        },
+        character(1)
+      ),
+      collapse = "\n"
+    ),
+    "\n\n"
+  )
 
-  if (exists("progress")) {
-    progress$set(message = "Getting R Code", value = 1)
-    progress$close()
+  str_filter <- teal.slice::get_filter_expr(datasets, datanames)
```

Comment:
@gogonzo @mhallal1 - this will be right, but I'm super confused about how this interacts with the merge expression which also has filtering in? No need to explain I guess as this all is being changed